cirq-core/cirq/sim/state_vector_simulator.py

```diff
@@ -199,11 +188,11 @@ def state_vector(self):
                 |  6  |   1    |   1    |   0    |
                 |  7  |   1    |   1    |   1    |
         """
-        return self._final_simulator_state.state_vector.copy()
+        return self.final_state_vector.copy()
```

Comment:
I need to think about this change a bit - there have been several issues in the past regarding whether or not this copies the final state.

Reply:
It would have to be this way to preserve `result.state_vector() is not result.state_vector()`, if anyone depends on that. IDK why we have two functions that return the state vector anyway. I'd support deprecating this one.

Reply:
Oh, we need this one for `StateVectorMixin` support. Hmm, that said I don't particularly like the preemptive copy here. Especially in a `TrialResult` where the simulation is already done, it seems like overkill.

Reply:
I'm going to leave this as is. This is the same behavior as we have currently. That said, I think we should remove it eventually, but in a separate PR.